client/blocks/login/social.jsx

```diff
@@ -110,6 +129,14 @@ class SocialLoginForm extends Component {
 	}
 
 	render() {
+		const { redirectTo } = this.props;
+		// If calypso is loaded in a popup, we don't want to open a second popup for social login
+		// let's use the redirect flow instead in that case.
+		//const isPopup = typeof window !== 'undefined' && window.opener && window.opener !== window;
```

Comment:
Also lets add here a check if browser is safari, because of: https://github.com/google/google-api-javascript-client/issues/297#issuecomment-333869742